src/platform/plugins/shared/discover/public/application/main/components/tabs_view/use_preview_data.ts

```diff
@@ -86,23 +97,66 @@ const getPreviewQuery = (query: TabPreviewData['query'] | undefined): TabPreview
 
   return {
     ...query,
-    query: trimmedQuery || DEFAULT_PREVIEW_QUERY.query,
+    query: trimmedQuery || (dataViewName ? '' : DEFAULT_PREVIEW_QUERY.query),
   };
 };
 
+const getDataViewNameFromInitialInternalState = (
+  initialInternalState: TabState['initialInternalState'] | undefined,
+  savedDataViews: DataViewListItem[]
+): string | undefined => {
+  if (!initialInternalState?.serializedSearchSource) {
+    return undefined;
+  }
+
+  const index = initialInternalState.serializedSearchSource.index;
+
+  if (typeof index === 'string') {
+    const matchedDataView = savedDataViews.find((dv) => dv.id === index);
+    return matchedDataView?.name || matchedDataView?.title;
+  }
+
+  if (index?.name) {
+    return index.name;
+  }
+
+  return undefined;
+};
+
+const getPreviewTitle = (
+  query: TabPreviewData['query'] | undefined,
+  dataViewName: string | undefined
+) => {
+  if (isOfAggregateQueryType(query)) {
+    return undefined;
+  }
+
+  return dataViewName
+    ? i18n.translate('discover.tabsView.tabPreviewDataViewTitle', {
+        defaultMessage: 'Data View: {dataViewName}',
```

Comment:
```suggestion
        defaultMessage: 'Data view: {dataViewName}',
```
Tiny nit: "Data view" should be sentence case here.